### ISSUE TITLE
margo_cfg memory leak in margo_init_opt_json() path

### ISSUE THREAD

**Issue by carns** [mochi-hpc/mochi-margo#65]
In GitLab by @carns on Oct 13, 2020, 08:11

There is a small memory leak associated with the parsed json in the margo_init_opt_json() path specifically:

```
==176576== 388 (72 direct, 316 indirect) bytes in 1 blocks are definitely lost in loss record 8 of 8
==176576==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==176576==    by 0x4C81F3F: json_object (value.c:61)
==176576==    by 0x4AC65C4: __mochi_cfg_get_component (mochi-cfg.c:256)
==176576==    by 0x4AC6B98: mochi_cfg_get_component (mochi-cfg.c:174)
==176576==    by 0x484FE49: margo_init_opt_json (margo.c:279)
==176576==    by 0x484FD10: margo_init_opt (margo.c:245)
==176576==    by 0x10ABF2: main (margo-example-server.c:40)
```

Sub-objects (for argobots and mercury) are using borrowed references that do not need to be decremented, but something else isn't quite right.

**Comment by carns** [comment]
In GitLab by @carns on Oct 13, 2020, 08:24

This was a mochi-cfg bug, fixed in mochi-cfg 0.2

**Comment by carns** [comment]
In GitLab by @carns on Oct 13, 2020, 08:24

closed
